browsers/node16.14.2-slim-chrome103-ff99/Dockerfile

```diff
@@ -0,0 +1,82 @@
+# WARNING: this file was autogenerated by generate-browser-image.js
+# using
+#   yarn add:browser -- 16.14.2-slim --chrome=103.0.5060.53 --firefox=99.0.1
+#
+# build this image with command
+#   docker build -t cypress/browsers:node16.14.2-slim-chrome103-ff99 .
+#
+#
+FROM cypress/base:16.14.2-slim
+
+USER root
+
+RUN node --version
+
+COPY ./global-profile.sh /tmp/global-profile.sh
+RUN cat /tmp/global-profile.sh >> /etc/profile && rm /tmp/global-profile.sh
+
+# Install dependencies
+RUN apt-get update && \
+  apt-get install -y \
+  fonts-liberation \
+  git \
+  libcurl4 \
+  libcurl3-gnutls \
+  libcurl3-nss \
+  xdg-utils \
+  wget \
+  curl \
+  # firefox dependencies
+  bzip2 \
+  # add codecs needed for video playback in firefox
+  # https://github.com/cypress-io/cypress-docker-images/issues/150
+  mplayer \
+  \
+  # clean up
+  && rm -rf /var/lib/apt/lists/* \
+  && apt-get clean
+
+# install libappindicator3-1 - not included with Debian 11
+RUN wget --no-verbose /usr/src/libappindicator3-1_0.4.92-7_amd64.deb "http://ftp.us.debian.org/debian/pool/main/liba/libappindicator/libappindicator3-1_0.4.92-7_amd64.deb" && \
+  dpkg -i /usr/src/libappindicator3-1_0.4.92-7_amd64.deb ; \
+  apt-get install -f -y && \
+  rm -f /usr/src/libappindicator3-1_0.4.92-7_amd64.deb
+
+# install Chrome browser
+RUN node -p "process.arch === 'arm64' ? 'Not downloading Chrome since we are on arm64: https://crbug.com/677140' : process.exit(1)" || \
```

Comment:
> you could be running a docker image on 'arm64' that was built on 'x64' and would have the chrome, linux and firefox install.

Yeah, this is possible. Today, this is what happens on M1 because there aren't `linux/arm64` builds, so Docker falls back to `linux/amd64`.

Side note, I wasn't able to get an `amd64` Chrome build running inside of a `linux/amd64` Docker image on an M1 Mac, the QEMU layer fails - apparently this is a known issue: https://bugs.chromium.org/p/chromium/issues/detail?id=677140#c15

> Do we build the docker images on both 'arm64' and 'x64'?

Yes, we build and test  in `arm64` and `x64` in CI with this PR. Then, both images are re-built on an `x64` image and pushed to Hub at the same time. This is the only way I could figure out how to do this in a way that results in a clean Docker Hub image list: https://cypressio.slack.com/archives/C033S1JBPV0/p1656514666620389

> Does docker figure out which to download? Do we need to publish the images separately? 

Yeah, Docker will figure it out automatically - you can still force Docker to install a specific platform with `--platform`, but otherwise, Docker will check out the image's manifest and install the most appropriate architecture. So if you look at my test `base` images here, you'll see each tag has two digests available, one for each platform: https://hub.docker.com/r/flotwig/base/tags

Reply:
@flotwig, thanks for answering my questions!